Result/Result.swift

```diff
@@ -179,6 +179,15 @@ public func ?? <T, Error> (left: Result<T, Error>, @autoclosure right: () -> Res
 	return left.recoverWith(right())
 }
 
+// MARK: - Derive result from failable closure
+
+public func materialize<T>(f: () throws -> T) -> Result<T, ErrorType> {
```

Comment:
It's better to keep type info, but how is `E` inferred? Currently, we can't know what kind of error `() throws -> T` throws. And if we can't get `E` from `() throws -> T`, type annotation is required to call this function as below:

```swift
let data = NSData()
let result: Result<AnyObject, NSError> = materialize {
    try NSJSONSerialization.JSONObjectWithData(data, options: [])
}
```